### DISCUSSION TITLE
Inertia indices

### DISCUSSION THREAD

**Discussion by alexdegt-ops** [gap-system/gap#5691] [Q&A] [answered]
Does anyone know if there is a built-in function in GAP to find the inertia indices of a symmetric matrix, say, over Integers or Rationals? (Couldn't find in the manual, but some functions, like DirectSumMat, are still not documented.) Of course I have something written "on the knee" but apparently it's too slow: for a (4600×4600)-integer matrix it works slower than SmithNormalForm, which does not seem normal.

**Comment by fingolfin** [comment, marked as answer]
Caveat: I never heard the term "inertia index of a matrix" before (probably because I never took an english course in relevant topics, and we called it something else in my language), but some googling suggest that this might refer to a triple $(n_+,n_0,n_-)$ defined e.g. [here](https://en.wikipedia.org/wiki/Sylvester%27s_law_of_inertia). Assuming that this is indeed what you mean, then I am not aware of anything in GAP computing these values. In some sense, computing this kind of thing is not very "natural" in GAP (no judgement implied here, it just is different from what "typical" code does, as means to explain why I think it is unlikely that there is a hidden implementation lurking in some package -- though of course it is possible). In general linear algebra in characteristic 0 in GAP is not very good compared to the state of the art.

You may be better of using some other system for this. 

Or if you really want to have this in GAP, you could try to implement it yourself. E.g. this sketches a basic algorithm for computing the inertia with a link to a more elaborate algorithm: https://math.stackexchange.com/questions/142549

**Comment by alexdegt-ops** [comment]
Yes, I mean that, and of course I have it implemented by myself. It’s probably very clumsy as one of my very first exercises in `GAP` following what I learned about matrices in high school but, as long as it works, I don’t care to optimize. I just find it strange that so basic linear algebra is not implemented whereas much more advanced/expensive things, like Smith normal form or `ShortestVectors`, are. But it’s certainly matter of taste and priorities, so I don’t complain. (In fact, last time I checked it wasn’t implemented in `Maple`, either 😊 Apparently, nowadays only topologists and algebraic geometers care about the signatures of their forms!)
